### PR TITLE
fix(ui): guard setup wizard when model catalog is empty

### LINKED ISSUE #108: Setup wizard reaches provider selection with zero provider choices
## Problem

The setup wizard has no boundary guard for an empty model catalog. If the bundled LiteLLM snapshot is missing/unreadable and live LiteLLM model loading fails, `get_available_models()` can return `{}`. `run_setup_wizard()` then calls `select_provider()` with that empty mapping.

`select_provider()` builds provider choices only from providers present in `available_models`. With an empty catalog, it reaches `Prompt.ask()` with an empty `choices` list.

This is structural debt because setup flow responsibilities are split incorrectly: model-catalog availability is a required precondition for provider selection, but no layer enforces it before the interactive prompt.

## Evidence

The structural audit traced the setup flow through `src/notewise/ui/setup_wizard.py`.

`get_available_models()` falls back through the catalog-loading paths around `src/notewise/ui/setup_wizard.py:179-185`.

The LiteLLM fallback path can return an empty normalized catalog on exception around `src/notewise/ui/setup_wizard.py:215-220`:

```py
except Exception:
    return _normalize_available_models({})
```

Existing tests already document this possibility. `tests/unit/ui/test_setup_wizard.py:309-321` verifies `get_available_models()` can return `{}` on LiteLLM failure.

`run_setup_wizard()` then calls provider selection unconditionally around `src/notewise/ui/setup_wizard.py:461-469`:

```py
available_models = get_available_models()
provider = select_provider(available_models)
```

`select_provider()` builds choices only for providers present in `available_models` around `src/notewise/ui/setup_wizard.py:260-295`:

```py
providers = []
for provider_key, provider_config in PROVIDER_CONFIG.items():
    if provider_key in available_models:
        providers.append(...)
...
Prompt.ask(..., choices=[str(i) for i in range(1, len(providers_list) + 1)])
```

When `available_models == {}`, the choices list is empty.

## Impact

`notewise setup` can fail or become unusable when model metadata is unavailable.

Trigger scenarios:

- Bundled model snapshot is missing or corrupt.
- LiteLLM import/model loading fails.
- Offline or constrained environment prevents fallback model discovery.

Blast radius:

- Setup cannot guide the user to a provider.
- The user sees an interactive prompt failure instead of an actionable setup error.
- Config writing may be partially attempted in future changes if no boundary guard is added.

## Suggested Fix

Add an explicit no-model/no-provider boundary before `select_provider()`.

Example:

```py
available_models = get_available_models()
if not available_models:
    console.print("No setup-safe model catalog is available. Reinstall NoteWise or retry with network access.")
    return current_config

provider = select_provider(available_models)
```

The setup wizard should not call `select_provider()` or `save_config()` when no providers are available.

## Acceptance Criteria

- [ ] `run_setup_wizard(force=True)` with `get_available_models()` returning `{}` does not call `select_provider()`.
- [ ] No config file is written or partially updated when no provider choices exist.
- [ ] User sees a clear setup failure/recovery message.
- [ ] Existing OAuth and API-key setup tests still pass.
- [ ] `select_provider()` is not responsible for handling an impossible empty catalog unless explicitly tested.

### PR DESCRIPTION
**Pull request by qorexdevs**
## Pull Request Checklist

- [x] I read `CONTRIBUTING.md` and followed the contribution workflow.
- [x] This PR targets `dev` (or is an explicit maintainer-managed release PR targeting `main`).
- [ ] I ran `make ci` locally and all checks passed (`format-check`, `lint-check`, `type-check`, `deps-check`, `security`, `test-cov`).
- [x] I added or updated tests for the behavior change.
- [ ] I updated docs for user-facing changes.
- [ ] I documented any breaking change in this PR description.

## Validation Commands

```bash
uv run pytest tests/unit/ui/test_setup_wizard.py -q
uv run pre-commit run --files src/notewise/ui/setup_wizard.py tests/unit/ui/test_setup_wizard.py
uv run pre-commit run --hook-stage pre-push --files src/notewise/ui/setup_wizard.py tests/unit/ui/test_setup_wizard.py
```

`make` is not available in this Windows environment, so I ran the equivalent pre-commit and test commands directly.

## Summary

Add an early guard in `run_setup_wizard()` for an empty setup model catalog.
When `get_available_models()` returns `{}`, the wizard now prints a recovery message and returns before provider/model prompts or config writes.
Added a unit test to verify `select_provider()` and `save_config()` are not called in this path.

## Related Issue

Closes #108

## Summary by Sourcery

Guard the setup wizard from running when no setup-safe models are available and ensure this behavior is covered by tests.

Bug Fixes:
- Prevent the setup wizard from proceeding when the model catalog is empty, returning the existing configuration and showing recovery guidance instead.

Tests:
- Add a unit test verifying the wizard exits early on an empty model catalog without prompting for a provider or saving config.

## Release Notes

* **Bug Fixes**
  * Setup wizard now validates model availability before proceeding with configuration and displays clear error messages if no models are found.

[Review Change Stack](https://app.coderabbit.ai/change-stack/whoisjayd/notewise/pull/117)

### CODE REVIEW

**Review by whoisjayd — CHANGES_REQUESTED** [review]
@qorexdevs please do suggested changes

File: src/notewise/ui/setup_wizard.py [hunk 1]
```diff
@@ -460,6 +460,15 @@ def run_setup_wizard(
 
     active_console.print("\n[cyan]Loading available models...[/cyan]")
     available_models = get_available_models(console=active_console)
+    if not available_models:
+        active_console.print(
+            "[red]No setup-safe model catalog is available right now.[/red]"
+        )
+        active_console.print(
+            "[yellow]Reinstall notewise or retry with network access, "
+            "then run setup again.[/yellow]"
```
[the remark this answers is not in the document]
Comment (qorexdevs) [reply src/notewise/ui/setup_wizard.py]: Done in 0ed955c: moved both setup-guard strings into src/notewise/_constants.py and now import/use them from setup_wizard.py.

File: src/notewise/ui/setup_wizard.py [hunk 2]
```diff
@@ -460,6 +460,15 @@ def run_setup_wizard(
...
+        )
+        return current_config
```
[the remark this answers is not in the document]
Comment (qorexdevs) [reply src/notewise/ui/setup_wizard.py:470]: I kept the current behavior here intentionally: this setup failure path returns existing config without writing, same as other early-exit wizard paths. It avoids turning a recoverable missing-catalog state into a hard failure for users running setup in constrained environments.

**Comment by whoisjayd** [comment]
@qorexdevs thanks for your contribution give repo a star if you find it useful
merging this pr
